### PR TITLE
Add waiting to cephcluster check

### PR DESCRIPTION
**Pull request by ebondare**
Fixes https://github.com/red-hat-storage/ocs-ci/issues/7747

### CODE REVIEW

**Comment by ebondare** [comment]
Successful test run: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/25567/console

File: ocs_ci/helpers/managed_services.py [hunk 1]
```diff
@@ -453,14 +462,15 @@ def verify_faas_provider_resources():
...
+        kind=constants.CEPH_CLUSTER,
+        namespace=config.ENV_DATA["cluster_namespace"],
+        resource_name=constants.CEPH_CLUSTER_NAME,
     )
-    cephcluster_yaml = cephcluster.get().get("items")[0]
-    log.info("Verifying that Cephcluster is Ready and hostNetworking is True")
-    assert (
-        cephcluster_yaml["status"]["phase"] == "Ready"
-    ), f"Status of cephcluster {cephcluster_yaml['metadata']['name']} is {cephcluster_yaml['status']['phase']}"
+    log.info("Waiting for Cephcluster to be Ready")
```
Comment (jilju) [inline ocs_ci/helpers/managed_services.py:470]: This is implemented differently in the function 'verify_provider_resources ' in this PR. Isn't that suitable here ?

File: ocs_ci/helpers/managed_services.py [hunk 2]
```diff
@@ -31,6 +31,15 @@
 log = logging.getLogger(__name__)
 
 
+class ResourceWithPhase(OCP):
```
Comment (jilju) [inline ocs_ci/helpers/managed_services.py:34]: _has_phase is declared as private variable in the class OCP. But many places outside the class alter it using "_has_phase=True". Eventhough this class helps to use the variable in an appropriate way, as this is not specific for managed services please move it to a different module.

File: ocs_ci/helpers/managed_services.py (hunk 1 above)
Comment (ebondare) [reply ocs_ci/helpers/managed_services.py:462]: I think using wait_for_phase function is preferable to re-implementing it using TimeoutSampler.

**Review by fbalak — APPROVED** [review]

**Review by jilju — APPROVED** [review]

**Comment by ebondare** [comment]
new successful deployment job with this PR: https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/1753/

**Pull request merged**